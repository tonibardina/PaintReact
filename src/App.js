import React, { Component } from 'react'
import Tools from './components/Tools'
import './style/App.css'

class App extends Component {
  constructor (props) {
    super(props)
    this.mouse = {
      x: 0, 
      y: 0,
      pressed: false
    }
    this.state = {
      undo_list: [],
      redo_list: [],
    }
  }

  componentDidMount () {
    this.context.lineWidth = 10
    this.context.lineCap = 'round'
    this.context.lineJoin = 'round'
  }

  handleMove = (e) => {
    this.mouse.x = e.pageX - this.canvas.offsetLeft
    this.mouse.y = e.pageY - this.canvas.offsetTop
    if (this.mouse.pressed) {
      this.context.lineTo(this.mouse.x, this.mouse.y)
      this.context.stroke()
    }
  }

  handleMouseDown = () => {
    let file = this.canvas.toDataURL()
    this.mouse.pressed = true
    this.context.beginPath()
    this.context.moveTo(this.mouse.x, this.mouse.y)
    this.saveState(file)
  }

  handleMouseUp = () => {
    this.mouse.pressed = false
  }

  handleMouseOut = () => {
    this.mouse.pressed = false
  }

  changeColor = (color) => {
    this.context.strokeStyle = color
  }

  changeLineWidth = (value) => {
    this.context.lineWidth = value
  }

  undo = () => {
    this.restoreState(this.state.undo_list, this.state.redo_list)
  }

  redo = () => {
    this.restoreState(this.state.redo_list, this.state.undo_list)
  }

  saveState = (file, list, keep_redo) => {
    keep_redo = keep_redo || false
    if(!keep_redo) {
      this.setState({
        redo_list: [],
      })
    }
    (list || this.state.undo_list).push(file)
  }

  restoreState = (pop, push) => {
    let file = this.canvas.toDataURL()
    const width = this.canvas.width 
    const height = this.canvas.height
    const context = this.context
    if(pop.length) {
      this.saveState(file, push, true)
      let restore_state = pop.pop()
      let image = new Image()
      image.src = restore_state
      image.onload = function () {
        context.clearRect(0, 0, width, height)
        context.drawImage(image, 0, 0, width, height, 0, 0, width, height)
      }
    }
  }

  clearWorkspace = () => {
    const width = this.canvas.width
    const height = this.canvas.height
    this.setState({
      undo_list: [],
      redo_list: []
    })
    this.context.clearRect(0, 0, width, height)
  }

  render () {
    return (
      <div style={{padding: 0}} className='container-fluid'>
        <Tools
          /*downloadFile={this.canvas}*/
          undo={this.undo}
          redo={this.redo}
          changeColor={this.changeColor} 
          changeLineWidth={this.changeLineWidth}
          clearWorkspace={this.clearWorkspace}
        />
        <div 
        className='canvasContainer' 
        style={{height: '100%', width: '100%'}} 
        ref={(s) => {
              if (s) {
                this.sketch = s
                this.canvasWidth = s.style.width
                this.canvasHeight = s.style.height
              }
          }}
        >
        <canvas
          ref={(c) => {
            if (c) {
              this.canvas = c
              this.context = c.getContext('2d')
            }
          }}
          id='canvas'
          style={{height: '100%', width: '100%'}}
          onMouseMove={this.handleMove}
          onMouseOver={this.handleMouseOver} 
          onMouseDown={this.handleMouseDown}
          onMouseUp={this.handleMouseUp}
          onMouseOut={this.handleMouseOut}
        />
      </div>
      </div>
    )
  }
}

export default App
