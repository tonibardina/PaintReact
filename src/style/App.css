canvas#canvas {
  cursor: crosshair;
}

.fa-arrow-right {
  color: white;
  background-color: white;
}

