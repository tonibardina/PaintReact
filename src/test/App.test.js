import React from 'react'
import renderer from 'react-test-renderer'
import {shallow, mount} from 'enzyme'
import Adapter from './setUpTests'

import App from '../App'

describe('App', () => {

  it('should build successfully', () => {
   
    const component = renderer.create(
      <App
      />
    )
    expect(component.toJSON()).toMatchSnapshot()
  
  })

  it('handle move successfully', () => {

    let move = false

    function handleMove () {
      move = true
    }

    const wrapper = shallow(
      <canvas
        id='canvas'
        onMouseMove={handleMove}
      />
    )
    wrapper.find('#canvas').simulate('mousemove')
    expect(move).toBe(true)

  })

  it('handle mousedown successfully', () => {

    let mouseDown = false

    function handleMouseDown () {
      mouseDown = true
    }

    const wrapper = shallow(
      <canvas
        id='canvas'
        onMouseDown={handleMouseDown}
      />
    )
    wrapper.find('#canvas').simulate('mousedown')
    expect(mouseDown).toBe(true)
    
  })

  it('handle mouseUp successfully', () => {

    let mouseUp = false

    function handleMouseUp () {
      mouseUp = true
    }

    const wrapper = shallow(
      <canvas
        id='canvas'
        onMouseUp={handleMouseUp}
      />
    )
    wrapper.find('#canvas').simulate('mouseUp')
    expect(mouseUp).toBe(true)
    
  })

  it('handle mouseOut successfully', () => {

    let mouseOut = false

    function handleMouseOut () {
      mouseOut = true
    }

    const wrapper = shallow(
      <canvas
        id='canvas'
        onMouseOut={handleMouseOut}
      />
    )
    wrapper.find('#canvas').simulate('mouseOut')
    expect(mouseOut).toBe(true)
    
  })

})