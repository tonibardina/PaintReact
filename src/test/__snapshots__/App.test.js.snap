// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should build successfully 1`] = `
<div
  className="container-fluid"
  style={
    Object {
      "padding": 0,
    }
  }
>
  <div>
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <a
            className="navbar-brand"
            id="title"
            style={
              Object {
                "color": "red",
              }
            }
          >
            Paint
          </a>
        </div>
        <ul
          className="nav navbar-nav"
          role={null}
        >
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              id="undo"
              name="undo"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-undo"
              />
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              id="redo"
              name="redo"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-repeat"
              />
            </a>
          </li>
          <li
            className="dropdown"
            style={undefined}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href="#"
              id="color-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Palette
               
              <span
                className="caret"
              />
            </a>
            <ul
              aria-labelledby="color-dropdown"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  id="palette1"
                  name="Colors1"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Color variety
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  id="palette2"
                  name="Colors2"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  Colorhunt selection
                </a>
              </li>
            </ul>
          </li>
          <li
            className="dropdown"
            style={undefined}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href="#"
              id="color-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Color
               
              <span
                className="caret"
              />
            </a>
            <ul
              aria-labelledby="color-dropdown"
              className="dropdown-menu"
              role="menu"
            >
              <div
                className="block-picker "
                style={
                  Object {
                    "MozBorderRadius": "6px",
                    "MozBoxShadow": "0 1px rgba(0,0,0,.1)",
                    "OBorderRadius": "6px",
                    "OBoxShadow": "0 1px rgba(0,0,0,.1)",
                    "WebkitBorderRadius": "6px",
                    "WebkitBoxShadow": "0 1px rgba(0,0,0,.1)",
                    "background": "#fff",
                    "borderRadius": "6px",
                    "boxShadow": "0 1px rgba(0,0,0,.1)",
                    "msBorderRadius": "6px",
                    "msBoxShadow": "0 1px rgba(0,0,0,.1)",
                    "position": "relative",
                    "width": 170,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "borderColor": "transparent transparent #ff0000 transparent",
                      "borderStyle": "solid",
                      "borderWidth": "0 10px 10px 10px",
                      "height": "0px",
                      "left": "50%",
                      "marginLeft": "-10px",
                      "position": "absolute",
                      "top": "-10px",
                      "width": "0px",
                    }
                  }
                />
                <div
                  style={
                    Object {
                      "MozBorderRadius": "6px 6px 0 0",
                      "OBorderRadius": "6px 6px 0 0",
                      "WebkitBorderRadius": "6px 6px 0 0",
                      "WebkitJustifyContent": "center",
                      "alignItems": "center",
                      "background": "#ff0000",
                      "borderRadius": "6px 6px 0 0",
                      "display": "flex",
                      "height": "110px",
                      "justifyContent": "center",
                      "msBorderRadius": "6px 6px 0 0",
                      "position": "relative",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "18px",
                        "position": "relative",
                      }
                    }
                  >
                    #ff0000
                  </div>
                </div>
                <div
                  style={
                    Object {
                      "padding": "10px",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "marginRight": "-10px",
                      }
                    }
                  >
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#E8DAB2",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#E8DAB2"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#4F6D7A",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#4F6D7A"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#C0D6DF",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#C0D6DF"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#EAEAEA",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#EAEAEA"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#D44D5C",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#D44D5C"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#083452",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#083452"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#FF6B37",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#FF6B37"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#ECA400",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#ECA400"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#EAF8BF",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#EAF8BF"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#006992",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#006992"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#95FF9F",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#95FF9F"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#136F63",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#136F63"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#22AAA1",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#22AAA1"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#4CE0D2",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#4CE0D2"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#84CAE7",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#84CAE7"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#641318",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#641318"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#CB2631",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#CB2631"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#A31E27",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#A31E27"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "#B76D72",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="#B76D72"
                      />
                    </span>
                    <span
                      onBlur={[Function]}
                      onFocus={[Function]}
                    >
                      <div
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        style={
                          Object {
                            "MozBorderRadius": "4px",
                            "OBorderRadius": "4px",
                            "WebkitBorderRadius": "4px",
                            "background": "red",
                            "borderRadius": "4px",
                            "cursor": "pointer",
                            "float": "left",
                            "height": "22px",
                            "marginBottom": "10px",
                            "marginRight": "10px",
                            "msBorderRadius": "4px",
                            "outline": "none",
                            "position": "relative",
                            "width": "22px",
                          }
                        }
                        tabIndex={0}
                        title="red"
                      />
                    </span>
                    <div
                      style={
                        Object {
                          "clear": "both",
                        }
                      }
                    />
                  </div>
                  <div
                    style={
                      Object {
                        "position": "relative",
                      }
                    }
                  >
                    <input
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder={undefined}
                      spellCheck="false"
                      style={
                        Object {
                          "MozBorderRadius": "4px",
                          "MozBoxShadow": "inset 0 0 0 1px #ddd",
                          "OBorderRadius": "4px",
                          "OBoxShadow": "inset 0 0 0 1px #ddd",
                          "WebkitBorderRadius": "4px",
                          "WebkitBoxShadow": "inset 0 0 0 1px #ddd",
                          "border": "0px",
                          "borderRadius": "4px",
                          "boxShadow": "inset 0 0 0 1px #ddd",
                          "boxSizing": "border-box",
                          "color": "#666",
                          "fontSize": "12px",
                          "height": "22px",
                          "msBorderRadius": "4px",
                          "msBoxShadow": "inset 0 0 0 1px #ddd",
                          "outline": "none",
                          "padding": "0 7px",
                          "width": "100%",
                        }
                      }
                      value="#FF0000"
                    />
                  </div>
                </div>
              </div>
            </ul>
          </li>
          <li
            className="dropdown"
            style={undefined}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href="#"
              id="basic-nav-dropdown"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Line
               
              <span
                className="caret"
              />
            </a>
            <ul
              aria-labelledby="basic-nav-dropdown"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <img
                    alt="1px Line"
                    id="line1"
                    name={1}
                    onClick={[Function]}
                    src="1pxLine.png"
                    style={
                      Object {
                        "padding": 10,
                      }
                    }
                  />
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <img
                    alt="3px Line"
                    id="line3"
                    name={3}
                    onClick={[Function]}
                    src="3pxLine.png"
                    style={
                      Object {
                        "padding": 10,
                      }
                    }
                  />
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <img
                    alt="5px Line"
                    id="line5"
                    name={5}
                    onClick={[Function]}
                    src="5pxLine.png"
                    style={
                      Object {
                        "padding": 10,
                      }
                    }
                  />
                </a>
              </li>
              <li
                className=""
                role="presentation"
                style={undefined}
              >
                <a
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="menuitem"
                  tabIndex="-1"
                >
                  <img
                    alt="10px Line"
                    id="line10"
                    name={10}
                    onClick={[Function]}
                    src="10pxLine.png"
                    style={
                      Object {
                        "padding": 10,
                      }
                    }
                  />
                </a>
              </li>
            </ul>
          </li>
          <li
            className="clearButton"
            role="presentation"
            style={undefined}
          >
            <a
              href="#"
              id="clearButton"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Clear
            </a>
          </li>
          <li
            className=""
            role="presentation"
            style={undefined}
          >
            <a
              download="myAwesomePaint"
              href="#"
              id="downloadButton"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              <i
                aria-hidden="true"
                className="fa fa-download"
              />
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <canvas
    height={768}
    id="canvas"
    onMouseDown={[Function]}
    onMouseMove={[Function]}
    onMouseOut={[Function]}
    onMouseOver={undefined}
    onMouseUp={[Function]}
    width={1024}
  />
</div>
`;
